node/test/Test/Pos/Client/Txp/Mode.hs

```diff
@@ -0,0 +1,288 @@
+                                                 suchThat)
+import           Test.QuickCheck.Monadic        (PropertyM, monadic)
+
+import           Pos.AllSecrets                 (mkInvAddrSpendingData, mkInvSecretsMap)
+import           Pos.Block.Core                 (Block, BlockHeader)
+import           Pos.Block.Slog                 (mkSlogContext)
+import           Pos.Block.Types                (Undo)
+import           Pos.Client.Txp.Addresses       (MonadAddresses (..))
+import           Pos.Client.Txp.Util            (TxCreateMode)
+import           Pos.Core                       (AddrSpendingData (..), HasCoreConstants,
+                                                 IsHeader, SlotId, Timestamp (..),
+                                                 makePubKeyAddressBoot)
+import           Pos.Crypto                     (SecretKey, toPublic)
+import           Pos.DB                         (MonadGState (..))
+import qualified Pos.DB                         as DB
+import qualified Pos.DB.Block                   as DB
+import           Pos.DB.DB                      as DB
+import           Pos.DB.DB                      (initNodeDBs)
+import           Pos.Genesis                    (GenesisContext (..), GenesisUtxo,
+                                                 GenesisWStakeholders,
+                                                 genesisContextImplicit, gtcUtxo,
+                                                 gtcWStakeholders)
+import qualified Pos.GState                     as GS
+import           Pos.Lrc                        (LrcContext (..), mkLrcSyncData)
+import           Pos.Slotting                   (HasSlottingVar (..), MonadSlots (..),
+                                                 MonadSlotsData, SimpleSlottingVar,
+                                                 mkSimpleSlottingVar)
+import qualified Pos.Slotting                   as Slot
+import           Pos.Ssc.Class                  (SscBlock)
+import           Pos.Ssc.Class.Helpers          (SscHelpersClass)
+import           Pos.Ssc.GodTossing             (SscGodTossing)
+import           Pos.Util.Util                  (Some, postfixLFields)
+
+-- Need Emulation because it has instance Mockable CurrentTime
+import           Test.Pos.Block.Logic.Emulation (Emulation (..), runEmulation)
+import           Test.Pos.Block.Logic.Mode      (genSuitableStakeDistribution)
+import           Test.Pos.Client.Txp.Util       (generateAddressAux, seedSize)
+
+----------------------------------------------------------------------------
+-- TestParams
+----------------------------------------------------------------------------
+
+-- An object of this type has instance Arbitrary and is used as the source
+-- of randomness in the tests.
+data TxpTestParams = TxpTestParams
+    { _ttpGenesisContext :: !GenesisContext
+    , _ttpStartTime      :: !Microsecond
+    } deriving (Show)
+
+makeClassy ''TxpTestParams
+
+instance Arbitrary TxpTestParams where
```

Comment:
This is almost perfect copypaste of `TestParams`, which you actually never use in tests - because you generate a new utxo every time manually, and you don't need information about genesis stakeholders at all. It seems to me that this may be safely removed completely - correct me if I'm wrong.

Reply:
Actually, I do use it: it is required by `initNodeDBs` in `initTxpTestContext`. In turn, `initNodeDBs` is required to have the `MonadGState` context.